src/cc/export/helpers.h

```diff
@@ -141,6 +141,24 @@ struct _name##_table_t { \
 __attribute__((section("maps/perf_output"))) \
 struct _name##_table_t _name = { .max_entries = 0 }
 
+// Table for pushing custom events to userspace via ring buffer
+#define BPF_RINGBUF_OUTPUT(_name) \
```

Comment:
I was experimenting with this earlier and it seems like most values cause map creation to fail with -EINVAL. Do you have some insight on why this might be the case?

Reply:
I've added a commit that allows the user to specify max_entries by number of pages.

Reply:
yes, it has to be in multiples of pages. I'd stick to bytes, though, because that's what kernel expects. If you want to be helpful to users (at the expense of keeping them in the dark about what's actually happening), you can automatically round-up to the closest multiple of page_size.

Reply:
@anakryiko What I'm doing right now is something like `BPF_RING_OUTPUT(buffer, 8)` which results in `{.max_entries = (8 * PAGE_SIZE)}`. I think this should be fine as an interface, as this resembles what we do for `PERF_OUTPUT` on the python side (user specifies number of pages when calling `open_perf_buffer`).

Reply:
So the kernel still sees bytes, but user simply provides number of pages.

Reply:
I'm worried that it just deviates from what's happening when people use libbpf instead of BCC and would complicate, unnecessarily, migration process.

Reply:
Not sure that I agree with this. I feel like it's simple enough to change to bytes when switching to use libbpf, and using bytes instead of pages here would not only confuse new users (when they can't figure out why their maps aren't working), but would also represent a significant departure for people who are already familiar with perf buffers in bcc, which deal in pages.

Reply:
@yonghong-song Do you have any thoughts on this?

Reply:
I don't feel very strongly on this, so feel free to stick to page count. But I'd love users to at least give it a passing thought that in case of PERFBUF it's pages of data per-CPU vs for RINGBUF it's one common ringbuf.

Reply:
I'll be certain to mention this when I write the documentation.